test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/FileHelper.kt

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+package com.bentley.imodeljs_test_app
+
+import android.content.ContentResolver
+import android.content.Context
+import android.net.Uri
+import android.provider.OpenableColumns
+import java.io.File
+import java.io.FileOutputStream
+import java.io.InputStream
+
+/**
+ * File related utility functions.
+ */
+object FileHelper {
+    /**
+     * Determines the display name for the uri using the content resolver.
+     * @param uri The input Uri.
+     * @param contentResolver The content resolver to query for the display name.
+     * @return The display name or null if it wasn't found.
+     */
+    fun getFileDisplayName(uri: Uri, contentResolver: ContentResolver): String? {
+        contentResolver.query(uri, null, null, null, null, null)?.let { cursor ->
+            if (cursor.moveToFirst()) {
+                val columnIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+                if (columnIndex >= 0) {
+                    return cursor.getString(columnIndex)
+                }
+            }
+            cursor.close()
```

Comment:
If close is needed for "not found", it seems like it should also be needed before the return on line 30. If it's not needed before that return, it seems like it shouldn't be needed here either.

Reply:
I completely reworked the functions as extension functions and moved them into MainActivity. Calling `close` is always required, Kotlin has a nice generic top-level `use` function that calls `close` for you in a finally block.